lib/elm/compiler.rb

```diff
@@ -23,16 +23,22 @@ def elm_executable_exists?(elm_make_path)
         false
       end
 
-      def compile_to_string(elm_make_path, elm_files)
+      def compile_to_string(elm_make_path, elm_files, debug)
         Tempfile.open(['elm', '.js']) do |tempfile|
-          elm_make(elm_make_path, elm_files, tempfile.path)
+          elm_make(elm_make_path, elm_files, tempfile.path, debug)
           return File.read tempfile.path
         end
       end
 
-      def elm_make(elm_make_path, elm_files, output_path)
-        Open3.popen3({"LANG" => "en_US.UTF8" }, elm_make_path, *elm_files, '--yes', '--output', output_path) do |_stdin, _stdout, stderr, wait_thr|
-          fail CompileError, stderr.gets(nil) if wait_thr.value.exitstatus != 0
+      def elm_make(elm_make_path, elm_files, output_path, debug)
+        if debug
+          Open3.popen3({"LANG" => "en_US.UTF8" }, elm_make_path, *elm_files, '--yes', '--output', output_path, '--debug') do |_stdin, _stdout, stderr, wait_thr|
+            fail CompileError, stderr.gets(nil) if wait_thr.value.exitstatus != 0
+          end
+        else
+          Open3.popen3({"LANG" => "en_US.UTF8" }, elm_make_path, *elm_files, '--yes', '--output', output_path) do |_stdin, _stdout, stderr, wait_thr|
```

Comment:
I think you could avoid a lot of this duplication by doing something like:

```
args = [{"LANG" => "en_US.UTF8" }, elm_make_path, *elm_files, '--yes', '--output', output_path]
args << "--debug" if debug
```

And then calling `popen3` with `args`.